<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

# Exercise 1

In [1]:
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

Just press the square box at the side!

# Exercise 2

In [2]:
print = "PRINT"

In [3]:
print("Hello World")

TypeError: 'str' object is not callable

Restarted the kernel!

In [1]:
print("Hello World")

Hello World


# Exercise 3

In [7]:
squares = [i**2 for i in range(10)]
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [8]:
%%timeit
squares = [i**2 for i in range(10)]

868 ns ± 26.1 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [9]:
squares = list()
for i in range(10):
    squares.append(i**2)
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [6]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

1.1 μs ± 18.1 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


The first set of code's ```%%timeit``` output implies that from 7 runs of 1,000,000 loops each, it took a mean time of 868 ns, with a standard deviation of 26.1 ns, for the code to be executed. 

The second set of code's ```%%timeit``` output implies that from 7 runs of 1,000,000 loops each, it took a mean time of 1.1 μs, with a standard deviation of 18.1 ns, for the code to be executed.

The results imply that the first set of code takes a shorter time to execute than the second set of code, which could be due to the fact that list comprehensions do not need to reference an already created list and append the integer to that list.

## Footnotes